🐛 Correct properties for attachment model

s0 opened the pull request:
Stricter types can be used for objectType and type, which was the case
for the old attachment model, so we should continue that here.

s0 committed:
🐛 Correct properties for attachment model

Stricter types can be used for objectType and type, which was the case
for the old attachment model, so we should continue that here.

Pl217 reviewed (changes requested):
Since, we're enriching attachment definition, I would take the opportunity to remove unused variable `COST_ATTACHMENT_VALUE` that is duplicated from `attachmentVersion.ts`, where it is actually used.  Both were added in f59ddc16959086e5c47ff6fe7a4a24eeafa69900

Can I take this PR to sneak in a commit that removes [this code comment](https://github.com/UN-OCHA/hpc-api-core/blob/develop/src/db/models/planVersion.ts#L47) since it [doesn't belong](https://github.com/UN-OCHA/hpc-api-core/pull/30#discussion_r741829569) after the changes we did today in 948d97d44d2e4eb8659953e824579ea3d62becf0?

Pl217 commented:
Also, what are your plans for [HPC-8205 branch](https://github.com/UN-OCHA/hpc-api-core/tree/HPC-8205) that removes `COST_ATTACHMENT_VALUE` (which I asked above, before realizing this :smile:) and also makes changes to `attachmentPrototype` and `attachmentVersion`?

s0 commented:
> Since, we're enriching attachment definition, I would take the opportunity to remove unused variable `COST_ATTACHMENT_VALUE` that is duplicated from `attachmentVersion.ts`, where it is actually used. Both were added in [f59ddc1](https://github.com/UN-OCHA/hpc-api-core/commit/f59ddc16959086e5c47ff6fe7a4a24eeafa69900)
> 
> Can I take this PR to sneak in a commit that removes [this code comment](https://github.com/UN-OCHA/hpc-api-core/blob/develop/src/db/models/planVersion.ts#L47) since it [doesn't belong](https://github.com/UN-OCHA/hpc-api-core/pull/30#discussion_r741829569) after the changes we did today in [948d97d](https://github.com/UN-OCHA/hpc-api-core/commit/948d97d44d2e4eb8659953e824579ea3d62becf0)?

sounds good to me

> Also, what are your plans for [HPC-8205 branch](https://github.com/UN-OCHA/hpc-api-core/tree/HPC-8205) that removes `COST_ATTACHMENT_VALUE` (which I asked above, before realizing this 😄) and also makes changes to `attachmentPrototype` and `attachmentVersion`?

That's still a WIP that i need to evolve a bit more, but i'll just rebase that ontop of whatever other changes we have in `hpc-api-core`, and will only finalize it when the related code in `hpc_service` is final and ready too, which I hope to have by end of day.

s0 committed:
🔥 Remove unused io-ts definition

s0 committed:
📝 Remove irrelevant comment

This comment was meant to "justify" why 'enum' was used
for kind property when projectVersion got defined in
f3fdb876bce22ca2785e96ebd3bb9dfa3f83f50a

It was corrected later in 948d97d44d2e4eb8659953e824579ea3d62becf0
to use 'checked' kind with `t.keyof()` io-ts type

s0 commented:
@Pl217 done

Pl217 committed:
🔖 Bump version to `v0.2.1`

Pl217 reviewed (approved):
I have also bumped version here, because I will publish a new version once this PR is merged, since changes from #25 are needed for https://github.com/UN-OCHA/hpc_service/pull/2445
